duracloud/management-console
# Add non-AWS configuration options

# What does this Pull Request do?
This PR adds additional configuration options in the AMA, so that one can configure non-AWS backend options.

When configuring Global Properties or Duracloud Mill, users now have the option to use:
- RabbitMQ instead of SNS for notifications
- RabbitMQ instead of SQS for the audit queue
- S3+Swift as a storage option for audit logs

Users now have the option of setting "spring" for email notifications, in which case an SMTP server is used instead of SES.

We have also included an update for the SQL schema. The schema should be updated before attempting to run the compiled WAR, even if you are not using the non-AWS config options.

Finally, we include the required update to the POM so that it uses HTTPS for Maven.

# How should this be tested?

This cannot be compiled without the changes that we made to the Duracloud repo, which will be in a forthcoming PR. Ideally, this and durastore/duradmin should be tested in tandem.

Pre-requisites for testing:

- An SMTP server
- A RabbitMQ server with:
  - `rabbitmq_message_timestamp` plugin installed
  - a fanout exchange

1. Build the AMA WAR in the usual way.
2. Ensure the database schema has been updated with `update-non-aws.sql`
3. Edit `notification` section in `config.properties` file with:
```
notification.type=spring
notification.host=smtp.example.com
notification.port=25
notification.user='exampleuser'
notification.pass='examplepass'
notification.from-address=duracloud@example.com
notification.admin-address=admin@example.com
```
4. Restart tomcat with new WAR
5. Log in to Root Console and edit Duracloud Mill parameters with new options
6. Edit Global Properties parameters with new options. Use the name of the fanout exchange as the exchange name
7. Restart tomcat. You should see messages that we have successfully connected to the SMTP server, and RabbitMQ has been selected for notifications. You should also see that queues have been created in RabbitMQ and they have been bound to the fanout exchange.

# Interested parties
@bbranan
